libweb3jsonrpc/Eth.cpp

```diff
@@ -359,17 +359,51 @@ string Eth::eth_sendRawTransaction(std::string const& _rlp)
 {
 	try
 	{
-		if (client()->injectTransaction(jsToBytes(_rlp, OnFailed::Throw)) == ImportResult::Success)
-		{
-			Transaction tx(jsToBytes(_rlp, OnFailed::Throw), CheckTransaction::None);
-			return toJS(tx.sha3());
-		}
-		else
-			return toJS(h256());
+		// Don't need to check the transaction signature (CheckTransaction::None) since it will
+		// be checked as a part of transaction import
+		Transaction t(jsToBytes(_rlp, OnFailed::Throw), CheckTransaction::None);
+		client()->importTransaction(t);
+		return toJS(t.sha3());
+	}
+	catch (InvalidSignature&)
```

Comment:
Can we get rid of duplication between `eth_sendRawTransaction` and `eth_sendTransaction` by catching `Exception` in both and having a helper like `string transactionExceptionToErrorMessage(Exception const&)` ?